crates/subspace-farmer/src/bin/subspace-farmer/main.rs

```diff
@@ -121,17 +122,117 @@ enum Subcommand {
     },
 }
 
+#[derive(Debug)]
+struct DiskFarm {
+    /// Path to directory where plots are stored, typically HDD.
+    plot_directory: PathBuf,
+    /// Path to directory for storing metadata, typically SSD.
+    metadata_directory: PathBuf,
+    /// How much space in bytes can farm use for plots (metadata space is not included)
+    allocated_plotting_space: u64,
+}
+
+impl FromStr for DiskFarm {
+    type Err = String;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        let parts = s.split(',').collect::<Vec<_>>();
+        if parts.len() != 3 {
+            return Err("Must contain 3 coma-separated components".to_string());
+        }
+
+        let mut plot_directory = None;
+        let mut metadata_directory = None;
+        let mut allocated_plotting_space = None;
+
+        for part in parts {
+            let part = part.splitn(2, '=').collect::<Vec<_>>();
```

Comment:
Is it fine that our parsing will fail if unix users will have path with `,` or `=`?

Reply:
I think yes, otherwise we need to handle character escaping and such, doesn't seem to be worth it, those characters not super common both directories where plots will be stored.